### PR TITLE
Merge websites-preview branch for BUILD

### PR DESCRIPTION
**Pull request by markcowl**
## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by markcowl** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/771/

File: src/ResourceManager/Websites/AzureRM.Websites.psd1 [hunk 1]
```diff
@@ -140,6 +141,8 @@ PrivateData = @{
         # External dependent modules of this module
         # ExternalModuleDependencies = @()
 
+        Prerelease = 'preview'
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/AzureRM.Websites.psd1]: This should be reverted.

File: src/ResourceManager/Websites/ChangeLog.md [hunk 2]
```diff
@@ -32,6 +32,10 @@
 * Fixed issue with importing aliases
 * New-AzureRMWebApp - added parameter set for simplified WebApp creation, with local git repository support.
 
+## Version 5.0.0-preview
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/ChangeLog.md]: Move the content of this into ## Current Release

File: src/ResourceManager/Websites/Commands.Websites.Test/Commands.Websites.Test.csproj [hunk 3]
```diff
@@ -84,6 +83,12 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.3\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/Commands.Websites.Test.csproj]: Remove clientruntime and system.management.automation references, as these are in common dependencies.

File: src/ResourceManager/Websites/Commands.Websites.Test/Commands.Websites.Test.csproj [hunk 4]
```diff
@@ -323,6 +310,4 @@
   </ItemGroup>
   <ItemGroup />
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-  <Import Project="..\..\..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets" Condition="Exists('..\..\..\packages\Microsoft.Bcl.Build.1.0.14\tools\Microsoft.Bcl.Build.targets')" />
-  <Import Project="..\..\..\..\tools\Common.Dependencies.Test.targets" />
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/Commands.Websites.Test.csproj]: Add this back.

File: src/ResourceManager/Websites/Commands.Websites.Test/packages.config [hunk 5]
```diff
@@ -6,14 +6,16 @@
...
-  <package id="Microsoft.Azure.Management.Websites" version="1.2.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Websites" version="1.7.1-preview" targetFramework="net452" />
   <package id="Microsoft.Azure.Test.Framework" version="1.0.6179.26854-prerelease" targetFramework="net45" />
   <package id="Microsoft.Azure.Test.HttpRecorder" version="1.6.7-preview" targetFramework="net45" />
   <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net45" />
   <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net45" />
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.10" targetFramework="net452" />
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/packages.config]: Remove these two references

File: src/ResourceManager/Websites/Commands.Websites/Commands.Websites.Netcore.csproj [hunk 6]
```diff
@@ -12,22 +12,28 @@
...
     <AssemblyOriginatorKeyFile>MSSharedLibKey.snk</AssemblyOriginatorKeyFile>
     <DefineConstants>TRACE;RELEASE;NETSTANDARD;SIGN</DefineConstants>
   </PropertyGroup>
 
-  <ItemGroup >
-    <PackageReference Include="Microsoft.Azure.Management.Websites" Version="1.6.0-preview" />
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Azure.Management.ResourceManager" Version="1.6.0-preview" />
+    <PackageReference Include="Microsoft.Azure.Management.Websites" Version="1.7.1-preview" />
+    <PackageReference Include="System.Collections.Specialized" Version="4.3.0" />
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites/Commands.Websites.Netcore.csproj]: Do we need all these package references?

File: tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv [hunk 7]
```diff
@@ -1144,6 +1144,9 @@
...
 "Microsoft.Azure.Commands.Subscription.dll","Microsoft.Azure.Commands.Subscription.Cmdlets.GetAzureRmSubscriptionDefinition","Get-AzureRmSubscriptionDefinition","0","1000","The cmdlet 'Get-AzureRmSubscriptionDefinition' has been removed and no alias was found for the original cmdlet name.","Add the cmdlet 'Get-AzureRmSubscriptionDefinition' back to the module, or add an alias to the original cmdlet name."
+"Microsoft.Azure.Commands.Resources.dll","Microsoft.Azure.Commands.Resources.GetAzureRoleDefinitionCommand","Get-AzureRmRoleDefinition","0","2000","The cmdlet 'Get-AzureRmRoleDefinition' no longer supports the parameter 'AtScopeAndBelow' and no alias was found for the original parameter name.","Add the parameter 'AtScopeAndBelow' back to the cmdlet 'Get-AzureRmRoleDefinition', or add an alias to the original parameter name."
```
Comment (maddieclayton) [inline tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv]: These three entries probably should be removed, as they aren't included in this PR.

File: src/ResourceManager/Websites/AzureRM.Websites.psd1 [hunk 8]
```diff
@@ -12,7 +12,7 @@
 # RootModule = ''
 
 # Version number of this module.
-ModuleVersion = '4.2.2'
+ModuleVersion = '5.0.0'
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/AzureRM.Websites.psd1]: What version do we want this module to be? Having a preview and a stable module with the same version could be confusing and might cause problems for installation - do we want this to bump to 6.0.0?

File: src/ResourceManager/Websites/Commands.Websites.Test/Commands.Websites.Test.csproj [hunk 9]
```diff
@@ -84,6 +83,9 @@
       <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\..\packages\Microsoft.IdentityModel.Clients.ActiveDirectory.2.28.3\lib\net45\Microsoft.IdentityModel.Clients.ActiveDirectory.WindowsForms.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime.Azure, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/Commands.Websites.Test.csproj]: Remove this clientruntime.azure reference and the system.management.automation reference below.

File: src/ResourceManager/Websites/Commands.Websites.Test/packages.config [hunk 10]
```diff
@@ -6,14 +6,15 @@
...
-  <package id="Microsoft.Azure.Management.Websites" version="1.2.0-preview" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Websites" version="1.7.1-preview" targetFramework="net452" />
   <package id="Microsoft.Azure.Test.Framework" version="1.0.6179.26854-prerelease" targetFramework="net45" />
   <package id="Microsoft.Azure.Test.HttpRecorder" version="1.6.7-preview" targetFramework="net45" />
   <package id="Microsoft.Bcl" version="1.1.10" targetFramework="net45" />
   <package id="Microsoft.Bcl.Async" version="1.0.168" targetFramework="net45" />
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.29" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.10" targetFramework="net452" />
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/Commands.Websites.Test/packages.config]: Remove this reference as well.

File: tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv [hunk 11]
```diff
@@ -1144,6 +1144,8 @@
...
 "Microsoft.Azure.Commands.Subscription.dll","Microsoft.Azure.Commands.Subscription.Cmdlets.NewAzureRmSubscriptionDefinition","New-AzureRmSubscriptionDefinition","0","1000","The cmdlet 'New-AzureRmSubscriptionDefinition' has been removed and no alias was found for the original cmdlet name.","Add the cmdlet 'New-AzureRmSubscriptionDefinition' back to the module, or add an alias to the original cmdlet name."
 "Microsoft.Azure.Commands.Subscription.dll","Microsoft.Azure.Commands.Subscription.Cmdlets.GetAzureRmSubscriptionDefinition","Get-AzureRmSubscriptionDefinition","0","1000","The cmdlet 'Get-AzureRmSubscriptionDefinition' has been removed and no alias was found for the original cmdlet name.","Add the cmdlet 'Get-AzureRmSubscriptionDefinition' back to the module, or add an alias to the original cmdlet name."
+"Microsoft.Azure.Commands.Resources.dll","Microsoft.Azure.Commands.Resources.GetAzureRoleDefinitionCommand","Get-AzureRmRoleDefinition","0","1050","The parameter set 'RoleDefinitionNameParameterSet' for cmdlet 'Get-AzureRmRoleDefinition' has been removed.","Add parameter set 'RoleDefinitionNameParameterSet' back to cmdlet 'Get-AzureRmRoleDefinition'."
```
Comment (maddieclayton) [inline tools/StaticAnalysis/Exceptions/BreakingChangeIssues.csv]: This still has two extra lines that shouldn't be here.

File: src/ResourceManager/Websites/ChangeLog.md (hunk 2 above)
Comment (markcowl) [reply src/ResourceManager/Websites/ChangeLog.md]: Done

**Comment by maddieclayton** [comment]
on demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/774/

**Pull request merged**